### ISSUE TITLE
[BUG] Wireguard install failing on fresh Debian 10

### ISSUE THREAD

**Issue by kermitoo** [swizzin/swizzin#925]
### What happened?

This happens because installing wireguard tries to install a new kernel version with wireguard built-in (linux-image-5.10-rt-amd64 in this case)
Swizzin installer then tries to modprobe the module in the old kernel, which doesn't exist.

2 Solutions : 
- Install wireguard-dkms instead if the kernel doesn't have wireguard built-in
- Or require a reboot after installing new kernel

### Swizzin commit

448031a

### What OS are you using?

Debian 10 (Buster)

### What architecture is your OS?

amd64

### Relevant logs and output

```plaintext
[2m[Sun Nov 13 02:44:43][m [2m...	Performing installation of 2 apt packages
(wireguard qrencode)[m
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  linux-image-5.10-rt-amd64 linux-image-5.10.0-0.deb10.19-rt-amd64
  wireguard-tools
Suggested packages:
  linux-doc-5.10 debian-kernel-handbook openresolv | resolvconf
The following NEW packages will be installed:
  linux-image-5.10-rt-amd64 linux-image-5.10.0-0.deb10.19-rt-amd64 wireguard
  wireguard-tools
0 upgraded, 4 newly installed, 0 to remove and 0 not upgraded.
Need to get 54.9 MB of archives.
After this operation, 316 MB of additional disk space will be used.
Get:1 http://security.debian.org/debian-security buster/updates/main amd64 linux-image-5.10.0-0.deb10.19-rt-amd64 amd64 5.10.149-2~deb10u1 [54.8 MB]
Get:2 http://ftp.debian.org/debian buster-backports/main amd64 wireguard-tools amd64 1.0.20210223-1~bpo10+1 [87.7 kB]
Get:3 http://ftp.debian.org/debian buster-backports/main amd64 wireguard all 1.0.20210223-1~bpo10+1 [8,284 B]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 linux-image-5.10-rt-amd64 amd64 5.10.149-2~deb10u1 [1,080 B]
Fetched 54.9 MB in 1s (67.3 MB/s)
Selecting previously unselected package linux-image-5.10.0-0.deb10.19-rt-amd64.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 64409 files and directories currently installed.)
Preparing to unpack .../linux-image-5.10.0-0.deb10.19-rt-amd64_5.10.149-2~deb10u1_amd64.deb ...
Unpacking linux-image-5.10.0-0.deb10.19-rt-amd64 (5.10.149-2~deb10u1) ...
Selecting previously unselected package linux-image-5.10-rt-amd64.
Preparing to unpack .../linux-image-5.10-rt-amd64_5.10.149-2~deb10u1_amd64.deb ...
Unpacking linux-image-5.10-rt-amd64 (5.10.149-2~deb10u1) ...
Selecting previously unselected package wireguard-tools.
Preparing to unpack .../wireguard-tools_1.0.20210223-1~bpo10+1_amd64.deb ...
Unpacking wireguard-tools (1.0.20210223-1~bpo10+1) ...
Selecting previously unselected package wireguard.
Preparing to unpack .../wireguard_1.0.20210223-1~bpo10+1_all.deb ...
Unpacking wireguard (1.0.20210223-1~bpo10+1) ...
Setting up linux-image-5.10.0-0.deb10.19-rt-amd64 (5.10.149-2~deb10u1) ...
I: /vmlinuz.old is now a symlink to boot/vmlinuz-4.19.0-22-amd64
I: /initrd.img.old is now a symlink to boot/initrd.img-4.19.0-22-amd64
I: /vmlinuz is now a symlink to boot/vmlinuz-5.10.0-0.deb10.19-rt-amd64
I: /initrd.img is now a symlink to boot/initrd.img-5.10.0-0.deb10.19-rt-amd64
/etc/kernel/postinst.d/initramfs-tools:
update-initramfs: Generating /boot/initrd.img-5.10.0-0.deb10.19-rt-amd64
/etc/kernel/postinst.d/zz-update-grub:
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-5.10.0-0.deb10.19-rt-amd64
Found initrd image: /boot/initrd.img-5.10.0-0.deb10.19-rt-amd64
Found linux image: /boot/vmlinuz-4.19.0-22-amd64
Found initrd image: /boot/initrd.img-4.19.0-22-amd64
Found linux image: /boot/vmlinuz-4.19.0-21-amd64
Found initrd image: /boot/initrd.img-4.19.0-21-amd64
Warning: os-prober will be executed to detect other bootable partitions.
Its output will be used to detect bootable binaries on them and create new boot entries.
done
Setting up linux-image-5.10-rt-amd64 (5.10.149-2~deb10u1) ...
Setting up wireguard-tools (1.0.20210223-1~bpo10+1) ...
wg-quick.target is a disabled or a static unit, not starting it.
Setting up wireguard (1.0.20210223-1~bpo10+1) ...
Processing triggers for man-db (2.8.5-2) ...
Reading package lists...
Building dependency tree...
Reading state information...
The following package was automatically installed and is no longer required:
  linux-image-4.19.0-21-amd64
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  libqrencode4
The following NEW packages will be installed:
  libqrencode4 qrencode
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 78.0 kB of archives.
After this operation, 168 kB of additional disk space will be used.
Get:1 http://mirrors.online.net/debian buster/main amd64 libqrencode4 amd64 4.0.2-1 [38.2 kB]
Get:2 http://mirrors.online.net/debian buster/main amd64 qrencode amd64 4.0.2-1 [39.8 kB]
Fetched 78.0 kB in 0s (1,468 kB/s)
Selecting previously unselected package libqrencode4:amd64.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 69249 files and directories currently installed.)
Preparing to unpack .../libqrencode4_4.0.2-1_amd64.deb ...
Unpacking libqrencode4:amd64 (4.0.2-1) ...
Selecting previously unselected package qrencode.
Preparing to unpack .../qrencode_4.0.2-1_amd64.deb ...
Unpacking qrencode (4.0.2-1) ...
Setting up libqrencode4:amd64 (4.0.2-1) ...
Setting up qrencode (4.0.2-1) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for libc-bin (2.28-10+deb10u2) ...
[2m[Sun Nov 13 02:45:06][m [36mLOG	No failures in apt log found[m
[2m[Sun Nov 13 02:45:06][m 	[32m[2m✔   Apt install complete[m
modprobe: FATAL: Module wireguard not found in directory /lib/modules/4.19.0-22-amd64
[2m[Sun Nov 13 02:45:06][m [31m[1mERROR	Could not modprobe Wireguard, script will now terminate.[m
[2m[Sun Nov 13 02:45:06][m [31m	Please consult the above and/or check the log (less -R +G /root/logs/swizzin.log)[m
[2m[Sun Nov 13 02:45:06][m [1mINFO	Please ensure a kernel headers package is installed that matches the currently running kernel.
Currently running kernel:
4.19.0-22-amd64
Installed kernel headers: 


You may be able to resolve this error with `apt install linux-headers-4.19.0-22-amd64` or a system reboot. If you are using a custom kernel, your package names may differ.
Please consult the swizzin log for further info if required.[m
```

**Comment by zunami** [comment]
I have the same problem

**Comment by brettpetch** [comment]
You're running a custom kernel... you need the headers for said kernel.

**Comment by liaralabs** [comment]
@zunami It looks like you're running proxmox, which isn't an officially supported distro.

@kermitoo are you able to try the fx/wg_buster branch and let me know if this resolves it for you?

```
cd /etc/swizzin
git pull
git checkout wg/fix_buster
# maybe git pull again?
box install wireguard
```

When you are done testing you can simple revert back to the master branch:

```
cd /etc/swizzin
git checkout master
```

**Comment by kermitoo** [comment]
@liaralabs 

```
[Mon Feb 27 17:36:37] LOG     >>>> `box install wireguard`
[Mon Feb 27 17:36:37] LOG     git @ 9377c13 2>&1
[Mon Feb 27 17:36:37] INFO    Currently on branch: fix/wg_buster
[Mon Feb 27 17:36:37] INFO    Installing wireguard
[Mon Feb 27 17:36:37] DOCS    Further reference: https://swizzin.ltd/applications/wireguard
[Mon Feb 27 17:36:38] INPUT   Setup has detected that eno1 is your main interface, is this correct?
39188832-6b01-4e27-9934-0b5fdfa83e96
[Mon Feb 27 17:36:40] LOG     Inserted apt logcheck marker
[Mon Feb 27 17:36:40] LOG     Variables unset
[Mon Feb 27 17:36:40] LOG     New variables exported
[Mon Feb 27 17:36:40] LOG     Apt log age = 324, update is forced= true
[Mon Feb 27 17:36:40] ...     Performing apt update
Hit:1 http://security.debian.org/debian-security buster/updates InRelease
Hit:2 http://mirrors.online.net/debian buster InRelease
Reading package lists...
[Mon Feb 27 17:36:41] LOG     No failures in apt log found
[Mon Feb 27 17:36:41]         ✔   Done
a08c20fb-d337-46f7-b3a4-874d8db69bfd
[Mon Feb 27 17:36:41] LOG     Inserted apt logcheck marker
[Mon Feb 27 17:36:41] LOG     Variables unset
[Mon Feb 27 17:36:41] LOG     APT will install recommends
[Mon Feb 27 17:36:41] LOG     New variables exported
[Mon Feb 27 17:36:41] LOG     No dpkg locks found
Reading package lists...
Building dependency tree...
Reading state information...
[Mon Feb 27 17:36:41] LOG     Apt check returned succesfully
[Mon Feb 27 17:36:41] LOG     Apt log age = 0, update is forced= 
[Mon Feb 27 17:36:41] LOG      Apt sufficiently up to date, skipping
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package wireguard
[Mon Feb 27 17:36:42] ERROR   The simulation check resulted in an error. Please consult the log.
[Mon Feb 27 17:36:42]         Please consult the above and/or check the log (less -R +G /root/logs/swizzin.log)
```

**Comment by kermitoo** [comment]
https://github.com/swizzin/swizzin/blob/9377c1399d422810a1c7b4cfef6c5e76b4eb0cfb/scripts/install/wireguard.sh#L46

should be

` case $codename in `

then it is installing fine

**Comment by liaralabs** [comment]
Good catch, thanks for reviewing.

**Comment by kermitoo** [comment]
@liaralabs 

Oops, still new kernel was installed

```
[Mon Feb 27 17:55:47] LOG	>>>> `box install wireguard`
[Mon Feb 27 17:55:47] LOG	git @ 9377c13 2>&1
[Mon Feb 27 17:55:47] INFO	Currently on branch: fix/wg_buster
[Mon Feb 27 17:55:47] INFO	Installing wireguard
[Mon Feb 27 17:55:47] DOCS	Further reference: https://swizzin.ltd/applications/wireguard
[Mon Feb 27 17:55:47] INPUT	Setup has detected that eno1 is your main interface, is this correct?
[Mon Feb 27 17:55:49] INFO	Enabling Debian buster backports repository
19381c32-7938-4316-ae71-daa5b1d6a404
[Mon Feb 27 17:55:49] LOG	Inserted apt logcheck marker
[Mon Feb 27 17:55:49] LOG	Variables unset
[Mon Feb 27 17:55:49] LOG	New variables exported
[Mon Feb 27 17:55:49] LOG	Apt log age = 1148, update is forced= true
[Mon Feb 27 17:55:49] ...	Performing apt update
Hit:1 http://mirrors.online.net/debian buster InRelease
Get:2 http://ftp.debian.org/debian buster-backports InRelease [51.4 kB]
Hit:3 http://security.debian.org/debian-security buster/updates InRelease
Get:4 http://ftp.debian.org/debian buster-backports/main Sources [456 kB]
Get:5 http://ftp.debian.org/debian buster-backports/main amd64 Packages [487 kB]
Get:6 http://ftp.debian.org/debian buster-backports/main Translation-en [411 kB]
Fetched 1,406 kB in 2s (569 kB/s)
Reading package lists...
[Mon Feb 27 17:55:52] LOG	No failures in apt log found
[Mon Feb 27 17:55:52] 	✔   Done
500fd52f-c84c-4355-9288-a581a5c30b24
[Mon Feb 27 17:55:52] LOG	Inserted apt logcheck marker
[Mon Feb 27 17:55:52] LOG	Variables unset
[Mon Feb 27 17:55:52] LOG	New variables exported
[Mon Feb 27 17:55:52] LOG	Apt log age = 0, update is forced= true
[Mon Feb 27 17:55:52] ...	Performing apt update
Hit:1 http://mirrors.online.net/debian buster InRelease
Hit:2 http://ftp.debian.org/debian buster-backports InRelease
Hit:3 http://security.debian.org/debian-security buster/updates InRelease
Reading package lists...
[Mon Feb 27 17:55:54] LOG	No failures in apt log found
[Mon Feb 27 17:55:54] 	✔   Done
5ecdbd70-5298-4f37-b146-77f1b198f424
[Mon Feb 27 17:55:54] LOG	Inserted apt logcheck marker
[Mon Feb 27 17:55:54] LOG	Variables unset
[Mon Feb 27 17:55:54] LOG	APT will install recommends
[Mon Feb 27 17:55:54] LOG	New variables exported
[Mon Feb 27 17:55:54] LOG	No dpkg locks found
Reading package lists...
Building dependency tree...
Reading state information...
[Mon Feb 27 17:55:54] LOG	Apt check returned succesfully
[Mon Feb 27 17:55:54] LOG	Apt log age = 0, update is forced= 
[Mon Feb 27 17:55:54] LOG	 Apt sufficiently up to date, skipping
Reading package lists...
Building dependency tree...
Reading state information...
The following package was automatically installed and is no longer required:
  linux-image-4.19.0-21-amd64
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  dkms libqrencode4 linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64
  linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19
  wireguard-tools
Suggested packages:
  python3-apport menu openresolv | resolvconf
The following NEW packages will be installed:
  dkms libqrencode4 linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64
  linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19
  qrencode wireguard wireguard-dkms wireguard-tools
0 upgraded, 11 newly installed, 0 to remove and 0 not upgraded.
Inst dkms (2.6.1-4 Debian:10.13/oldstable [all])
Inst libqrencode4 (4.0.2-1 Debian:10.13/oldstable [amd64])
Inst linux-compiler-gcc-8-x86 (4.19.269-1 Debian-Security:10/oldstable [amd64])
Inst linux-headers-4.19.0-23-common (4.19.269-1 Debian-Security:10/oldstable [all])
Inst linux-kbuild-4.19 (4.19.269-1 Debian-Security:10/oldstable [amd64])
Inst linux-headers-4.19.0-23-amd64 (4.19.269-1 Debian-Security:10/oldstable [amd64])
Inst linux-headers-amd64 (4.19+105+deb10u18 Debian-Security:10/oldstable [amd64])
Inst qrencode (4.0.2-1 Debian:10.13/oldstable [amd64])
Inst wireguard-dkms (1.0.20210219-1~bpo10+1 Debian Backports:buster-backports [all])
Inst wireguard-tools (1.0.20210223-1~bpo10+1 Debian Backports:buster-backports [amd64])
Inst wireguard (1.0.20210223-1~bpo10+1 Debian Backports:buster-backports [all])
Conf dkms (2.6.1-4 Debian:10.13/oldstable [all])
Conf libqrencode4 (4.0.2-1 Debian:10.13/oldstable [amd64])
Conf linux-compiler-gcc-8-x86 (4.19.269-1 Debian-Security:10/oldstable [amd64])
Conf linux-headers-4.19.0-23-common (4.19.269-1 Debian-Security:10/oldstable [all])
Conf linux-kbuild-4.19 (4.19.269-1 Debian-Security:10/oldstable [amd64])
Conf linux-headers-4.19.0-23-amd64 (4.19.269-1 Debian-Security:10/oldstable [amd64])
Conf linux-headers-amd64 (4.19+105+deb10u18 Debian-Security:10/oldstable [amd64])
Conf qrencode (4.0.2-1 Debian:10.13/oldstable [amd64])
Conf wireguard-dkms (1.0.20210219-1~bpo10+1 Debian Backports:buster-backports [all])
Conf wireguard-tools (1.0.20210223-1~bpo10+1 Debian Backports:buster-backports [amd64])
Conf wireguard (1.0.20210223-1~bpo10+1 Debian Backports:buster-backports [all])
[Mon Feb 27 17:55:55] LOG	Simulation completed succesfully
[Mon Feb 27 17:55:55] ...	Performing installation of 3 apt packages
(wireguard qrencode wireguard-dkms)
Reading package lists...
Building dependency tree...
Reading state information...
The following package was automatically installed and is no longer required:
  linux-image-4.19.0-21-amd64
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  linux-image-5.10-rt-amd64 linux-image-5.10.0-0.deb10.21-rt-amd64
  wireguard-tools
Suggested packages:
  linux-doc-5.10 debian-kernel-handbook openresolv | resolvconf
The following NEW packages will be installed:
  linux-image-5.10-rt-amd64 linux-image-5.10.0-0.deb10.21-rt-amd64 wireguard
  wireguard-tools
0 upgraded, 4 newly installed, 0 to remove and 0 not upgraded.
Need to get 54.9 MB of archives.
After this operation, 317 MB of additional disk space will be used.
Get:1 http://security.debian.org/debian-security buster/updates/main amd64 linux-image-5.10.0-0.deb10.21-rt-amd64 amd64 5.10.162-1~deb10u1 [54.9 MB]
Get:2 http://ftp.debian.org/debian buster-backports/main amd64 wireguard-tools amd64 1.0.20210223-1~bpo10+1 [87.7 kB]
Get:3 http://ftp.debian.org/debian buster-backports/main amd64 wireguard all 1.0.20210223-1~bpo10+1 [8,284 B]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 linux-image-5.10-rt-amd64 amd64 5.10.162-1~deb10u1 [1,080 B]
Fetched 54.9 MB in 8s (7,298 kB/s)
Selecting previously unselected package linux-image-5.10.0-0.deb10.21-rt-amd64.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 68812 files and directories currently installed.)
Preparing to unpack .../linux-image-5.10.0-0.deb10.21-rt-amd64_5.10.162-1~deb10u1_amd64.deb ...
Unpacking linux-image-5.10.0-0.deb10.21-rt-amd64 (5.10.162-1~deb10u1) ...
Selecting previously unselected package linux-image-5.10-rt-amd64.
Preparing to unpack .../linux-image-5.10-rt-amd64_5.10.162-1~deb10u1_amd64.deb ...
Unpacking linux-image-5.10-rt-amd64 (5.10.162-1~deb10u1) ...
Selecting previously unselected package wireguard-tools.
Preparing to unpack .../wireguard-tools_1.0.20210223-1~bpo10+1_amd64.deb ...
Unpacking wireguard-tools (1.0.20210223-1~bpo10+1) ...
Selecting previously unselected package wireguard.
Preparing to unpack .../wireguard_1.0.20210223-1~bpo10+1_all.deb ...
Unpacking wireguard (1.0.20210223-1~bpo10+1) ...
Setting up wireguard-tools (1.0.20210223-1~bpo10+1) ...
wg-quick.target is a disabled or a static unit, not starting it.
Setting up linux-image-5.10.0-0.deb10.21-rt-amd64 (5.10.162-1~deb10u1) ...
I: /vmlinuz.old is now a symlink to boot/vmlinuz-4.19.0-23-amd64
I: /initrd.img.old is now a symlink to boot/initrd.img-4.19.0-23-amd64
I: /vmlinuz is now a symlink to boot/vmlinuz-5.10.0-0.deb10.21-rt-amd64
I: /initrd.img is now a symlink to boot/initrd.img-5.10.0-0.deb10.21-rt-amd64
/etc/kernel/postinst.d/initramfs-tools:
update-initramfs: Generating /boot/initrd.img-5.10.0-0.deb10.21-rt-amd64
/etc/kernel/postinst.d/zz-update-grub:
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-5.10.0-0.deb10.21-rt-amd64
Found initrd image: /boot/initrd.img-5.10.0-0.deb10.21-rt-amd64
Found linux image: /boot/vmlinuz-4.19.0-23-amd64
Found initrd image: /boot/initrd.img-4.19.0-23-amd64
Found linux image: /boot/vmlinuz-4.19.0-22-amd64
Found initrd image: /boot/initrd.img-4.19.0-22-amd64
Found linux image: /boot/vmlinuz-4.19.0-21-amd64
Found initrd image: /boot/initrd.img-4.19.0-21-amd64
Warning: os-prober will be executed to detect other bootable partitions.
Its output will be used to detect bootable binaries on them and create new boot entries.
done
Setting up linux-image-5.10-rt-amd64 (5.10.162-1~deb10u1) ...
Setting up wireguard (1.0.20210223-1~bpo10+1) ...
Processing triggers for man-db (2.8.5-2) ...
Reading package lists...
Building dependency tree...
Reading state information...
The following packages were automatically installed and are no longer required:
  linux-image-4.19.0-21-amd64 linux-image-4.19.0-22-amd64
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  libqrencode4
The following NEW packages will be installed:
  libqrencode4 qrencode
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 78.0 kB of archives.
After this operation, 168 kB of additional disk space will be used.
Get:1 http://mirrors.online.net/debian buster/main amd64 libqrencode4 amd64 4.0.2-1 [38.2 kB]
Get:2 http://mirrors.online.net/debian buster/main amd64 qrencode amd64 4.0.2-1 [39.8 kB]
Fetched 78.0 kB in 0s (1,245 kB/s)
Selecting previously unselected package libqrencode4:amd64.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 73652 files and directories currently installed.)
Preparing to unpack .../libqrencode4_4.0.2-1_amd64.deb ...
Unpacking libqrencode4:amd64 (4.0.2-1) ...
Selecting previously unselected package qrencode.
Preparing to unpack .../qrencode_4.0.2-1_amd64.deb ...
Unpacking qrencode (4.0.2-1) ...
Setting up libqrencode4:amd64 (4.0.2-1) ...
Setting up qrencode (4.0.2-1) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for libc-bin (2.28-10+deb10u2) ...
Reading package lists...
Building dependency tree...
Reading state information...
The following packages were automatically installed and are no longer required:
  linux-image-4.19.0-21-amd64 linux-image-4.19.0-22-amd64
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  dkms linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64
  linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19
Suggested packages:
  python3-apport menu
The following NEW packages will be installed:
  dkms linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64
  linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19
  wireguard-dkms
0 upgraded, 7 newly installed, 0 to remove and 0 not upgraded.
Need to get 11.6 MB of archives.
After this operation, 60.9 MB of additional disk space will be used.
Get:1 http://mirrors.online.net/debian buster/main amd64 dkms all 2.6.1-4 [74.4 kB]
Get:2 http://security.debian.org/debian-security buster/updates/main amd64 linux-compiler-gcc-8-x86 amd64 4.19.269-1 [635 kB]
Get:3 http://ftp.debian.org/debian buster-backports/main amd64 wireguard-dkms all 1.0.20210219-1~bpo10+1 [254 kB]
Get:4 http://security.debian.org/debian-security buster/updates/main amd64 linux-headers-4.19.0-23-common all 4.19.269-1 [8,596 kB]
Get:5 http://security.debian.org/debian-security buster/updates/main amd64 linux-kbuild-4.19 amd64 4.19.269-1 [868 kB]
Get:6 http://security.debian.org/debian-security buster/updates/main amd64 linux-headers-4.19.0-23-amd64 amd64 4.19.269-1 [1,119 kB]
Get:7 http://security.debian.org/debian-security buster/updates/main amd64 linux-headers-amd64 amd64 4.19+105+deb10u18 [7,048 B]
Fetched 11.6 MB in 3s (3,872 kB/s)
Selecting previously unselected package dkms.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 73667 files and directories currently installed.)
Preparing to unpack .../0-dkms_2.6.1-4_all.deb ...
Unpacking dkms (2.6.1-4) ...
Selecting previously unselected package linux-compiler-gcc-8-x86.
Preparing to unpack .../1-linux-compiler-gcc-8-x86_4.19.269-1_amd64.deb ...
Unpacking linux-compiler-gcc-8-x86 (4.19.269-1) ...
Selecting previously unselected package linux-headers-4.19.0-23-common.
Preparing to unpack .../2-linux-headers-4.19.0-23-common_4.19.269-1_all.deb ...
Unpacking linux-headers-4.19.0-23-common (4.19.269-1) ...
Selecting previously unselected package linux-kbuild-4.19.
Preparing to unpack .../3-linux-kbuild-4.19_4.19.269-1_amd64.deb ...
Unpacking linux-kbuild-4.19 (4.19.269-1) ...
Selecting previously unselected package linux-headers-4.19.0-23-amd64.
Preparing to unpack .../4-linux-headers-4.19.0-23-amd64_4.19.269-1_amd64.deb ...
Unpacking linux-headers-4.19.0-23-amd64 (4.19.269-1) ...
Selecting previously unselected package linux-headers-amd64.
Preparing to unpack .../5-linux-headers-amd64_4.19+105+deb10u18_amd64.deb ...
Unpacking linux-headers-amd64 (4.19+105+deb10u18) ...
Selecting previously unselected package wireguard-dkms.
Preparing to unpack .../6-wireguard-dkms_1.0.20210219-1~bpo10+1_all.deb ...
Unpacking wireguard-dkms (1.0.20210219-1~bpo10+1) ...
Setting up linux-compiler-gcc-8-x86 (4.19.269-1) ...
Setting up linux-headers-4.19.0-23-common (4.19.269-1) ...
Setting up linux-kbuild-4.19 (4.19.269-1) ...
Setting up dkms (2.6.1-4) ...
Setting up wireguard-dkms (1.0.20210219-1~bpo10+1) ...
Loading new wireguard-1.0.20210219 DKMS files...
Building for 4.19.0-23-amd64 5.10.0-0.deb10.21-rt-amd64
Building initial module for 4.19.0-23-amd64
Done.

wireguard.ko:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/4.19.0-23-amd64/updates/dkms/

depmod.......

DKMS: install completed.
Module build for kernel 5.10.0-0.deb10.21-rt-amd64 was skipped since the
kernel headers for this kernel does not seem to be installed.
Setting up linux-headers-4.19.0-23-amd64 (4.19.269-1) ...
Setting up linux-headers-amd64 (4.19+105+deb10u18) ...
Processing triggers for man-db (2.8.5-2) ...
[Mon Feb 27 17:57:03] LOG	No failures in apt log found
[Mon Feb 27 17:57:03] 	✔   Apt install complete
[Mon Feb 27 17:57:03] 	✔   Done
```

**Comment by liaralabs** [comment]
Sneaky sneaky. I think I finally figured out what's causing this behavior.

The way apt packages are looped through during installation means that the backports package (wireguard-dkms) is not considered as a proper way to resolve the missing wireguard module dependency issues presented when:

```
apt install wireguard
```

As the first package installed.

With the backports enabled, I get this output when requesting to install wireguard only -- the first step of the install command

```
root@debian10:/etc/swizzin# apt install wireguard-dkms
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  dkms linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64 linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19 wireguard
  wireguard-tools
Suggested packages:
  python3-apport menu
The following NEW packages will be installed:
  dkms linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64 linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19 wireguard
  wireguard-dkms wireguard-tools
0 upgraded, 9 newly installed, 0 to remove and 0 not upgraded.
Need to get 11.6 MB of archives.
After this operation, 61.2 MB of additional disk space will be used.
```


It appears the solution is to request installation of `wireguard-dkms` only:

```
root@debian10:/etc/swizzin# apt install wireguard-dkms
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  dkms linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64 linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19 wireguard
  wireguard-tools
Suggested packages:
  python3-apport menu
The following NEW packages will be installed:
  dkms linux-compiler-gcc-8-x86 linux-headers-4.19.0-23-amd64 linux-headers-4.19.0-23-common linux-headers-amd64 linux-kbuild-4.19 wireguard
  wireguard-dkms wireguard-tools
0 upgraded, 9 newly installed, 0 to remove and 0 not upgraded.
Need to get 11.6 MB of archives.
After this operation, 61.2 MB of additional disk space will be used.
```

**Comment by liaralabs** [comment]
Hopefully fixed once and for all with 6b007e1dde889758b88a7a98e0fc53542bac2c53
